python/GafferUI/ExpressionUI.py

```diff
@@ -332,7 +334,14 @@ def __dropText( self, widget, dragData ) :
 
 		return None
 
-	def __error( self, plug, source, error ) :
+	def __errorOnIdle( self, plug, source, error ) :
+		# An error in the expression could occur during a compute triggered by a repaint.
+		# ( For example, if a user uses an expression to drive Backdrop text )
+		# If we forced a repaint right away, this would be a recursive repaint which could cause
+		# a Qt crash, so we wait for idle.
+		QtCore.QTimer.singleShot( 0, functools.partial(
```

Comment:
The general idea in GafferUI is that we hide Qt as much as possible so we're free to change toolkits in the future (Gaffer actually started life with GTK), and also so there are hopefully fewer hoops to jump through when migrating between Qt versions. There are a couple of options here :

- Use `GafferUI.EventLoop.addIdleCallback()` rather than `QTimer`
- Use the `@GafferUI.LazyMethod` decorator.

The latter should require the least code, and is probably the most expressive of what we're trying to do here.